### DISCUSSION TITLE
Quick capture into "inbox"

### DISCUSSION THREAD

**Discussion by bepolymathe** [freeplane/freeplane#478] [General]
Hi all, 

In my daily practice of note taking (todo and notes) I use the GTD method which implies to write down very quickly the things that go through my head in an "inbox" folder. 

I would like to reproduce this behavior in Freeplane with the idea of a keyboard shortcut to quickly enter a new node in a parent node "inbox". 

What do you think would be the best way to do this?

**Comment by macmarrum** [comment]
Hi

Assuming that you have a single "inbox" and that the node has the style "Inbox" assigned, this script bound to a hot key will do the trick: https://github.com/macmarrum/freeplane-scripting/blob/main/src/scripts/newNodeInInbox.groovy

**Reply by bepolymathe** [reply]
Great i will try this. Thanks you so much.
It would be great to make it works with multiple maps 😃

**Reply by macmarrum** [reply]
Sure thing 😄 

> It would be great to make it works with multiple maps 😃

I wonder what you mean by that. It would help me if you could describe a scenario.

**Comment by bepolymathe** [comment]
Hi @macmarrum 

Yes of course it was not clear! I often work with one map for "journaling" and tasks and others open maps for each ongoing project. 

It would be really interesting to be able to capture a task or an idea to the inbox node (in the "journaling" map) from a project open in another tab. 

By the way, looking at your (impressive) collection of scripts I deduce that you rather use the node style to give a status to tasks (next, waiting, etc) 🤔. I tend to use attributes. I'd be interested if you could explain/show in more detail how you handle GTD in Freeplane. 😃

**Reply by EdoFro** [reply]
I use GTD myself (or a "customized" version of it) since a long time.
In this time, I have used attributes and styles to give status to tasks .

Finally, I opted to use styles. Here are the reasons why I choose them over attributes:

- Both are a ways to mark nodes as tasks and define their status (both are ok)
- it is very useful that you can easy see which nodes are tasks and their status. (styles are better in that than attributes)
- You can solve the previous point by creating conditional styles that show when a node has an attribute and its value, but that has an extra cost in speed that you can notice when working with large and complex maps

At the end I use styles to set tasks status, but when needed I also use attributes to add other information, like "owner" or "due date".

I'm very interested in this topic and I think we can get some "best Practices" compilation that can be useful as base information, when developing scripts, so that they are useful for more people and not only for a single user.

jm2c

edo

**Reply by bepolymathe** [reply]
Hi @EdoFro (and thanks for your super scripts/addon) !

Thank you for your answer. I was wondering because I would like to be able to use the richness of the filters. I think you are right. I am currently combining the two as well. 

I have styles for:  
- Todo 
- Todo checked

But I have attributes for: 
- Status: "next" or "waiting"
- Scheduled : "2022-06-30"
- Deadline : "2022-07-14"
- Project : "name of the project to which the todo belongs
- Effort : "high" or "low" or "medium"

Managing more with styles I guess you have styles for "waiting" or "done"?

Also, I'm curious if you put your tasks in projects? On my side, I do journaling and I like to leave the task in its creation context... but I wonder... 

![image](https://user-images.githubusercontent.com/29939805/176750801-87391028-806c-4d98-a2c4-bf0260330125.png)


Finally, I would love to be able to search my tasks more quickly based on due dates and planning. Of course, I can search like that but if I want the tasks of the day (or in 7 days) I have to change the dates by hand and redo the filter each time. 

![image](https://user-images.githubusercontent.com/29939805/176749980-4ce67f98-5fdf-4bf3-8b3c-b87c0beedc04.png)

Another question comes to me. How do you manage the capture of a task or a resource from a phone (e.g. Android)? Do you have a process that allows a quick or even automatic integration in Freeplane ?

**Comment by macmarrum** [comment]
> It would be really interesting to be able to capture a task or an idea to the inbox node (in the "journaling" map) from a project open in another tab.

Please feel free to try this one:
https://github.com/macmarrum/freeplane-scripting/blob/main/src/scripts/newNodeInInboxOfGtdMap.groovy
Just change the name of the mindmap in the script to reflect yours.

> By the way, looking at your (impressive) collection of scripts I deduce that you rather use the node style to give a status to tasks (next, waiting, etc) 🤔. I tend to use attributes. I'd be interested if you could explain/show in more detail how you handle GTD in Freeplane. 😃

In my mindmaps I usually capture notes for a project. Sometimes the nodes reflect a simple sequence of events. In any case, where there's a Next Action or a Waiting For, I add a node and use a [script](https://github.com/macmarrum/freeplane-scripting/blob/main/src/scripts/GTD_Next_Action.groovy) to set the style and to add a "created" or "closed" attribute.
[freeplaneUserGuide-GTD.zip](https://github.com/freeplane/freeplane/files/9024706/freeplaneUserGuide-GTD.zip)

To see my Next Actions and Waiting Fors, I use a filter.
I use a formula in the details of a node (e.g. root) to create a report that counts the NAs and WFs.

**Reply by bepolymathe** [reply]
Hi @macmarrum 

> Please feel free to try this one:

That’s great ! Thank you. 



> I add a node and use a [script](https://github.com/macmarrum/freeplane-scripting/blob/main/src/scripts/GTD_Next_Action.groovy) to set the style and to add a "created" or "closed" attribute.

Nice idea. Thanks for sharing. 

And so you don't use any date information in the tasks (deadline, planning)?

I hadn't necessarily thought about the formulas but I'll take a closer look.

**Reply by bepolymathe** [reply]
> Please feel free to try this one:
> https://github.com/macmarrum/freeplane-scripting/blob/main/src/scripts/newNodeInInboxOfGtdMap.groovy
> Just change the name of the mindmap in the script to reflect yours.

This script is very useful. Do you think it would be possible to adapt it slightly so that it also includes a link to the node where we were at the time of the capture?

**Reply by macmarrum** [reply]
> This script is very useful. 

😀 
> Do you think it would be possible to adapt it slightly so that it also includes a link to the node where we were at the time of the capture?

Here it is. 
https://github.com/macmarrum/freeplane-scripting/blob/main/src/scripts/newNodeInInboxOfGtdMapWithBacklink.groovy
The script respects `Preferences…->Environment->Hyperlink types->Links`, i.e. creates either a relative or an absolute link to the other mindmap.

**Reply by bepolymathe** [reply]
Thanks a lot ! That’s great !

**Comment by bepolymathe** [comment]
Hi, 

Thanks again for your help with the scripts. I'm coming up with something great in the workflow reorganization with a map that mixes journaling and GTD while linking to other project maps for documentation and document writing (thanks to MarkdownHelper and thanks to @EdoFro ). 

The advantage of creating the tasks from my journal notes is that I can find the source (what we were talking about) but also the date and the context of creation thanks to the links. It is very practical!

![image](https://user-images.githubusercontent.com/29939805/176988679-7c5a4dc1-75c8-4063-b297-fd7dc580842a.png)


Thanks a lot guys. 👍🏼 

PS: I would just miss eventually a script to automatically move the tasks marked with the style "done" to an "archives" node.

**Reply by macmarrum** [reply]
Excellent!
I think that freedom of choice is essential and Freeplane helps in that. Everyone can construct their own workflow and choose how they want their journals/projects/tasks to look like and behave.

> I would just miss eventually a script to automatically move the tasks marked with the style "done" to an "archives" node.

Depending on the workflow, I would probably go with something like this, to find in Inbox all Done and move them to Archive

```groovy
def inboxStyleName = 'Inbox'
def archiveStyleName = 'Archive'
def doneStyleName = 'Done'

def allNodes = c.findAll()
def inboxNode = allNodes.find { it.style.name == inboxStyleName }
def archiveNode = allNodes.find { it.style.name == archiveStyleName }

inboxNode.find { it.style.name == doneStyleName }.each { it.moveTo(archiveNode) }
```

Alternatively, a script that marks selected task as done and moves them to Archive

```groovy
// @ExecutionModes({ON_SINGLE_NODE})

def archiveStyleName = 'Archive'
def doneStyleName = 'Done'

def archiveNode = c.findAll().find { it.style.name == archiveStyleName }

c.selecteds.each {
  it.style.name = doneStyleName
  it.moveTo(archiveNode)
}
````

**Reply by bepolymathe** [reply]
> Alternatively, a script that marks selected task as done and moves them to Archive

Great ! This one would be great for me. But i have an error

![image](https://user-images.githubusercontent.com/29939805/177013043-7a108d95-1833-4333-9ae5-ad192caea750.png)

I need to start learn scripting !

**Reply by bepolymathe** [reply]
I didn't say anything! It was obvious... I had not created the "Archive" style. Sorry for the inconvenience. Thanks @macmarrum !

Anyway, it makes me want to look at what I could do in terms of scripting. I'm not a developer but I'm interested in playing ;-) 

As a researcher who works with a lot of diary or interview excerpts, I think I might be able to handle it with Freeplane 🤔

**Comment by bepolymathe** [comment]
Hi ! 

Now that I have a great workflow for tasks in Freeplane I would be curious how you capture your tasks on mobile when you are not behind a screen?

**Reply by euu2021** [reply]
In my Android, I capture using Todoist, because it has a quick capture tool that works very well. Then, once a day, I copy the notes from todoist into FreePlane.  It works fine for me.

**Reply by bepolymathe** [reply]
And you just copy/paste to freeplane or it’s a bit automatic ? I do the same with Signal "message to self"

**Reply by euu2021** [reply]
No, it's not automatic. It would be possible to create an automatization using the Todoist API, but I still did not try it, because the copy and paste is very fast (it's possible to copy the whole inbox once and then clean it)

**Reply by bepolymathe** [reply]
Ok, thanks for reply. And out of curiosity, nothing automatic either for capture from the browser ?

**Reply by euu2021** [reply]
> And out of curiosity, nothing automatic either for capture from the browser ?

No, I just manually copy.
For the desktop, I remember seeing scripts that make the capture easier, for example, [this one](https://docs.freeplane.org/scripting/Scripts_collection.html#monitor-clipboard).

**Reply by bepolymathe** [reply]
Interesting, thank you. It's true that a popup window that can be called from anywhere could be useful to quickly capture tasks, phone numbers or links...

**Reply by bepolymathe** [reply]
Just for note, it could be very interesting to use Espanso to capture from anywhere 🤔

**Reply by euu2021** [reply]
I once created [this script using AutoHotkey](https://sourceforge.net/p/freeplane/discussion/758437/thread/0c97a52260/), but it is too cumbersome to use.
